### PR TITLE
AST-3199 - Fix: Post title & Featured image is not visible while editing in elementor with banner layout 2.

### PR DESCRIPTION
**Pull request by yssambare12**
### Description

When we create any post using Elementor it will not display the post title and featured image.

### Screenshots

- To reproduce this issue follow this video: https://d.pr/v/88E0YZ

**Steps To Reproduce:** 

1. Install Astra & Elementor 
2. Enable banner layout 2
3. Edit any post in elementor edit and see on frontend.

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: inc/compatibility/class-astra-elementor.php [hunk 1]
```diff
@@ -78,16 +78,20 @@ public function __construct() {
...
 			/** @psalm-suppress InvalidArgument */ // phpcs:ignore Generic.Commenting.DocComment.MissingShort
-			$edit_mode = get_post_meta( $post_id, '_elementor_edit_mode', true ); 
+			$edit_mode = get_post_meta( $post_id, '_elementor_edit_mode', true );
+			// Check layout type and show page title markup
+			$astra_post_type      = strval( get_post_type() );
+			$astra_banner_control = 'ast-dynamic-single-' . esc_attr( $astra_post_type );
+			$astra_layout_type    = astra_get_option( $astra_banner_control . '-layout', 'layout-1' );
 
-			if ( $edit_mode && $edit_mode === 'builder' ) { 
+			if ( $edit_mode && $edit_mode === 'builder' && $astra_layout_type != 'layout-2' ) {
```
Comment (imnavanath) [inline inc/compatibility/class-astra-elementor.php:94]: What if layout-1 is selected and elementor is active ?

Comment (yssambare12) [reply inc/compatibility/class-astra-elementor.php]: @imnavanath Logic reapplied please check now.

File: inc/compatibility/class-astra-elementor.php [hunk 2]
```diff
@@ -83,12 +83,20 @@ public function __construct() {
...
 		 */
 		function astra_entry_header_class_custom( $classes ) {
-			$post_id = astra_get_post_id(); 
+			$post_id = astra_get_post_id();
 			/** @psalm-suppress InvalidArgument */ // phpcs:ignore Generic.Commenting.DocComment.MissingShort
-			$edit_mode = get_post_meta( $post_id, '_elementor_edit_mode', true ); 
+			$edit_mode = get_post_meta( $post_id, '_elementor_edit_mode', true );
 
-			if ( $edit_mode && $edit_mode === 'builder' ) { 
+			if ( $edit_mode && $edit_mode === 'builder' ) {
```
Comment (imnavanath) [inline inc/compatibility/class-astra-elementor.php:90]:
```suggestion
			if ( ( $edit_mode && $edit_mode === 'builder' ) || ( $edit_mode === 'builder' && $astra_layout_type == 'layout-2' ) ) {
```

Comment (yssambare12) [reply inc/compatibility/class-astra-elementor.php]: Updated.

File: inc/compatibility/class-astra-elementor.php [hunk 3]
```diff
@@ -83,12 +83,15 @@ public function __construct() {
 		 * @since 4.1.0
 		 */
 		function astra_entry_header_class_custom( $classes ) {
-			$post_id = astra_get_post_id(); 
-			/** @psalm-suppress InvalidArgument */ // phpcs:ignore Generic.Commenting.DocComment.MissingShort
-			$edit_mode = get_post_meta( $post_id, '_elementor_edit_mode', true ); 
+				$edit_mode = get_post_meta( astra_get_post_id(), '_elementor_edit_mode', true );
 
-			if ( $edit_mode && $edit_mode === 'builder' ) { 
+			if ( ( $edit_mode && $edit_mode === 'builder' ) || ( $edit_mode === 'builder' && $astra_layout_type == 'layout-2' ) ) {
```
Comment (imnavanath) [inline inc/compatibility/class-astra-elementor.php:88]: It will throw undefined variable PHP error for `$astra_layout_type`

Comment (yssambare12) [reply inc/compatibility/class-astra-elementor.php:89]: @imnavanath `$astra_layout_type` variable is defined before the conditional statement for avoiding undefined variable error. Please check.

**Review by imnavanath — APPROVED** [review]

File: inc/compatibility/class-astra-elementor.php [hunk 4]
```diff
@@ -76,19 +76,22 @@ public function __construct() {
...
 		function astra_entry_header_class_custom( $classes ) {
-			$post_id = astra_get_post_id(); 
-			/** @psalm-suppress InvalidArgument */ // phpcs:ignore Generic.Commenting.DocComment.MissingShort
-			$edit_mode = get_post_meta( $post_id, '_elementor_edit_mode', true ); 
-
-			if ( $edit_mode && $edit_mode === 'builder' ) { 
+			$edit_mode = get_post_meta( astra_get_post_id(), '_elementor_edit_mode', true );
+			$astra_layout_type = astra_get_option( 'ast-dynamic-single-' . get_post_type() . '-layout', 'layout-1' );
+		
+			if ( ( $edit_mode && $edit_mode === 'builder' ) || ( $edit_mode === 'builder' && $astra_layout_type == 'layout-2' ) ) {
```
Comment (imnavanath) [inline inc/compatibility/class-astra-elementor.php:89]:
```suggestion
			if ( ( $edit_mode && $edit_mode === 'builder' ) || ( $edit_mode === 'builder' && $astra_layout_type === 'layout-2' ) ) {
```

**Pull request merged**